gdc_client/upload/client.py

```diff
@@ -452,12 +459,10 @@ def multipart_upload(self):
                 self.upload_parts()
                 self.check_multipart()
 
-                # try again in case some parts failed
-                if self.ns.completed != self.total_parts:
-                    self.upload_parts()
```

Comment:
so there are two reasons why I removed it:

1. the retry is redundant
each part is already retried up to MAX_RETRIES (which is 10 at the moment), so retrying the whole operation again for the failed parts seems arbitrary/not very useful (if we wanted to, we could make it a configurable thing, where we could set it to retry X number of times instead of just once). 

2. The log is confusing with the second call to self.upload_parts 
with my changes, if we take a file broken into 72 parts, and one part fails on the first upload_parts() call,  complete() will throw the exception saying "Multipart upload failed for file {}: completed parts: {}, total parts: {}, please try to resume", which is caught in handle_multiparts() and logged. So this would say 71 parts are completed out of 72 completed parts. 

The upload_parts() function seems functional in the sense that it is only capturing how many parts are successful out of the total parts tried in this particular run, but the problem arises when we call upload_parts() a second time (the current implementation). The logic in upload_parts() basically selects the parts that fails and uses the number of failed parts as the new total parts. Since this second upload_parts() gets called again before complete() gets called, we'll see that with this example, we'll see in the log that 0 parts were completed out of 1 total part, which is misleading. The problem is we're trying to use upload_parts(), which isn't necessarily in charge of keeping track of the overall successful parts out of the completed parts, but just on a particular run, out of the parts we want to try, how many were successful.